oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsSoundCard.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Oshi (https://github.com/oshi/oshi)
+ * <p>
+ * Copyright (c) 2010 - 2018 The Oshi Project Team
+ * <p>
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * <p>
+ * Maintainers:
+ * dblock[at]dblock[dot]org
+ * widdis[at]gmail[dot]com
+ * enrico.bianchi[at]gmail[dot]com
+ * <p>
+ * Contributors:
+ * https://github.com/oshi/oshi/graphs/contributors
+ */
+package oshi.hardware.platform.windows;
+
+import oshi.hardware.common.AbstractSoundCard;
+import oshi.jna.platform.windows.WbemcliUtil;
+import oshi.util.platform.windows.WmiUtil;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Sound Card data obtained by Wmi Queries for Windows.
+ *
+ * @author : BilalAM
+ */
+public class WindowsSoundCard extends AbstractSoundCard {
+    public WindowsSoundCard(String kernelVersion, String name, String codec) {
+        super(kernelVersion, name, codec);
+    }
+
+    enum SoundCardName {
+        MANUFACTURER, NAME;
+    }
+
+    enum SoundCardKernel {
+        DRIVERPROVIDERNAME, DRIVERNAME, DRIVERVERSION, DEVICENAME;
+    }
+
+    private static Map<String, String> NAME_MAP = new HashMap<>();
+    private static final String driverQuery = "Win32_PnPSignedDriver";
+    private static String whereClause = "WHERE DeviceName LIKE ";
+
+    /**
+     * Runs the Win32_SoundDevice query only once and initializes
+     * a Map with Name as key and Manufacturer as Value
+     *
+     * Also calls the createClause() method to build our WHERE clause query only once
+     */
+    static {
+        WbemcliUtil.WmiQuery<SoundCardName> SOUND_CARD_QUERY = new WbemcliUtil.WmiQuery<>("Win32_SoundDevice",SoundCardName.class);
+        WbemcliUtil.WmiResult<SoundCardName> SOUND_CARD_QUERY_RESULT = WmiUtil.queryWMI(SOUND_CARD_QUERY);
+        for (int i = 0; i < SOUND_CARD_QUERY_RESULT.getResultCount(); i++) {
+            NAME_MAP.put(WmiUtil.getString(SOUND_CARD_QUERY_RESULT, SoundCardName.NAME, i)
+                    , WmiUtil.getString(SOUND_CARD_QUERY_RESULT, SoundCardName.MANUFACTURER, i));
+        }
+        createClause(NAME_MAP);
+
+    }
+
+    /**
+     * Creates our Win32_PnPSignedDevice query with the WHERE clause taking the attributes
+     * from our map.
+     * @param map
+     *       The map whose keys will be used inside the Where Clause.
+     */
+    private static void createClause(Map<String, String> map) {
+        boolean isEnd = true;
+        for (String key : map.keySet()) {
```

Comment:
Any reason you're avoiding using a `StringBuilder` here?  Seems the classic case for it rather than using `+=` with Strings.  Do a `new Stringbuilder(DRIVER_QUERY_BASE)` and then append `" WHERE (foo)"` (1st iteration) and `" OR (foo)"` (2nd+ iteration) inside the loop.

Reply:
Well the main reason of using += is that it is really not a large string to be using a StringBuilder (and I feel a bit comfortable using +=s for small strings like this one lol )

Reply:
I have never used `+=` for string concatenation as a matter of taste. :)   For short strings always adding fixed elements the `+` is just fine (as you've done elsewhere) but almost always if I end up adding to a string inside a loop I prefer a `StringBuilder`.  Again this is mostly preference/performance/style.